driver/network/local/local.go

```diff
@@ -136,14 +139,27 @@ func NewLocalNetwork(config *driver.NetworkConfig) (*LocalNetwork, error) {
 	wg.Wait()
 
 	// If starting the validators failed, the network startup should fail.
-	if errors.Join(errs...) != nil {
-		err := net.Shutdown()
-		if err != nil {
-			errs = append(errs, err)
-		}
-		return nil, errors.Join(errs...)
+	if err := errors.Join(errs...); err != nil {
+		return nil, errors.Join(err, net.Shutdown())
 	}
 
+	// Setup infrastructure for managing applications on the network.
+	connection, err := net.DialRandomRpc()
```

Comment:
I think this assumes that once the connection is created, it stays live during the app. life-cycle. It may often work in a local env. but it is not stable generally. I think AppContext should have access to network, and be able to dial a new connection once needed.

Reply:
Full access to the network is not required -- the `AppContext` does not need an interface to create nodes and network listeners. So I added a `AppContext` specific `RpcClientFactory` interface that only provides access to re-creating connections as needed. Networks are automatically implementing this interface.

I updated the `AppContext` to create the connection internally and to keep it, but I have not implemented some re-connection mechanism in case there is a connection failure. Let's look into that once we encounter such issues.